instrumentation/messaging/src/main/java/brave/messaging/MessageConsumerAdapter.java

```diff
@@ -0,0 +1,20 @@
+package brave.messaging;
+
+public interface MessageConsumerAdapter<Msg> extends MessageAdapter<Msg> {
+  /**
+   * Messaging operation semantics, e.g. pull, push, send, receive, etc.
+   */
+  String operation(Msg message);
+
+  /**
+   * Message identifier, e.g. kafka record key, jms message correlation id.
+   */
+  String identifier(Msg message);
+
+  /**
+   * Removes propagation context from Message context carrier.
+   */
+  void clearPropagation(Msg message);
```

Comment:
My personal preference would be to return the Message. Reasoning is that some libraries or coding choices may prefer immutable data structures where copying and returning a new object is required.